src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java

```diff
@@ -833,7 +840,51 @@ public ActionForward approve(ActionMapping mapping, ActionForm form, HttpServlet
         if (StringUtils.isNotBlank(iWantDocForm.getNewAdHocRoutePerson().getId())) {
             insertAdHocRoutePerson(mapping, iWantDocForm, request, response);
         }
-        return super.approve(mapping, form, request, response);
+        
+        IWantDocument iwantDoc = iWantDocForm.getIWantDocument();
+        boolean shouldAddNote = getFinancialSystemWorkflowHelperService().isAdhocApprovalRequestedForPrincipal(
+                iwantDoc.getDocumentHeader().getWorkflowDocument(), GlobalVariables.getUserSession().getPrincipalId()) && 
+                StringUtils.isNotBlank(iwantDoc.getCompleteOption());
+        LOG.debug("approve, shouldAddNote: {}", shouldAddNote);
+
+        ActionForward forward = super.approve(mapping, form, request, response);
+        
+        if (shouldAddNote) {
+            addCompleteOptionNote(iwantDoc);
+        }
+        
+        return forward;
+    }
+
+    private void addCompleteOptionNote(IWantDocument iwantDoc) {
+        String noteText;
+        Person loggedInUser = GlobalVariables.getUserSession().getPerson();
+        if (StringUtils.equalsIgnoreCase(iwantDoc.getCompleteOption(), "Y")) {
```

Comment:
I would suggest using a constant for the "Y" instead of a string literal.

Reply:
ok